### ISSUE TITLE
bug: cursor stuck on command-line window

### ISSUE THREAD

**Issue by gjf7** [vscode-neovim/vscode-neovim#2190]
### Check the following:

- [X] I have checked the behavior after setting "vscode-neovim.neovimClean" in VSCode settings.
- [X] I have read the vscode-neovim docs.
- [X] I have searched existing issues.

### Neovim version (nvim -v)

v0.10.0

### Operating system/version

MacOS 14.1.1

### Describe the bug

I've recently encountered a couple of problems with the command-line window:

Empty Command History:
When I try to open the command-line window by typing `q:` in a buffer with no command history, nothing happens. The window doesn't appear and can only move cursor by using mouse, I don't see any error messages. It's as if the command is being ignored entirely, then type `q:` again got `MessagesManager: E1292: Command-line window is already open`

Non-Empty Command History:
If there are previous commands in the history, typing q: does open the command-line window. However, once it's open, I can't move the cursor at all. The usual navigation keys (h, j, k, l) have no effect. It's like the cursor is completely stuck.
When this happens, I noticed two error messages in the log:

```text

2024-08-01 16:53:23.257 [info] MainController: Starting nvim: /opt/homebrew/bin/nvim -N --embed --cmd source /Users/haochen/.vscode/extensions/asvetliakov.vscode-neovim-1.18.6/runtime/vscode-neovim.vim --clean
2024-08-01 16:53:23.351 [info] MainController: Nvim info: {
  logFile: '/Users/haochen/.local/state/nvim/log',
  configDir: '/Users/haochen/.config/nvim',
  nvimVersion: {
    patch: 0,
    api_level: 12,
    api_prerelease: null,
    api_compatible: 0,
    build: null,
    prerelease: true,
    major: 0,
    minor: 10
  }
}
2024-08-01 16:53:27.580 [error] BufferManager: nvim_open_win: E11: Invalid in command-line window; <CR> executes, CTRL-C quits
2024-08-01 16:53:27.580 [error] BufferManager: Unable to determine neovim window id for editor, viewColumn: 1, docUri: vscode-neovim://3/
```

### Steps To Reproduce

type q: in normal mode

### Expected Behavior

Command-line window should works exactly same as neovim.

**Comment by gjf7** [comment]
@xiyaowong Hi there, I'm new here and interested in helping with this issue. Would it be okay for me to work on it? I'd love to make this my first contribution. Is there anything important I should know before starting? Any tips or guidance would be great. Thanks!

**Comment by xiyaowong** [comment]
> Hi there, I'm new here and interested in helping with this issue. Would it be okay for me to work on it? I'd love to make this my first contribution.

Always welcome contributions!

> Is there anything important I should know before starting? Any tips or guidance would be great. Thanks!

This is a difficult feature, and we currently don't have any good ideas for now.

- Similar to #476 #505
- Some thoughts. #2149
- The most relevant code at the moment. 

	https://github.com/vscode-neovim/vscode-neovim/blob/1f9249be2523c9d3468ee3ec5a0d539956dcf17f/src/buffer_manager.ts#L813-L944

**Comment by xiyaowong** [comment]
The cmdline-window is a simple and specific special window. You can implement a custom editor dedicated to the cmdline-window,

**Comment by theol0403** [comment]
Yes, supporting this is 100% doable (and was supported in the past - around 2020), but it requires significant work in the buffer manager. Essentially, support for external buffers need to be added. 

This can likely be implemented by simplifying and refactoring the buffer manager, reducing code while generalizing the features. However, you can also attempt to investigate why external buffers aren't triggering a new vscode window. 

https://github.com/vscode-neovim/vscode-neovim/issues/1262

**Comment by gjf7** [comment]
Thank you both for your guidance. I'll need some time to familiarize myself with the codebase and relevant APIs. Once I have a more concrete plan, I'll share it here for your feedback before proceeding with any actual coding.
